### PR TITLE
fix shell/node issue close #50

### LINKED ISSUE #50: windows support
why does fly cli use [shell runner](https://github.com/flyjs/fly/blob/master/bin/index.js#L1)?

[gulp use different approach](https://github.com/gulpjs/gulp/blob/master/bin/gulp.js#L1)

isnt it bad experience for windows users?

### PR DESCRIPTION
**Pull request by iamstarkov**
as titled and see related #50 issue

### CODE REVIEW

File: bin/index.js [hunk 1]
```diff
@@ -1,5 +1,4 @@
-#!/bin/sh
-":" //# comment; exec /usr/bin/env node --harmony --harmony_arrow_functions "$0" "$@"
+#!/usr/bin/env node
```
Comment (watilde) [inline bin/index.js]: Nice work! Also, it need to add `--harmony` for using `yieldables`.

``` diff
- #!/usr/bin/env node
+ #!/usr/bin/env node --harmony
```

Comment (iamstarkov) [inline bin/index.js]: yay, awsm.

Comment (iamstarkov) [inline bin/index.js]: fixed
screen shot 2015-07-22 at 16 48 40

**Comment by hzlmn** [comment]
Could you please make it as one commit?

**Comment by iamstarkov** [comment]
@drKraken yep

**Comment by ghost** [comment]
Before this could land, everyone please make sure you understand that

``` sh
":" //# comment; exec /usr/bin/env node --harmony --harmony_arrow_functions "$0" "$@"
```

Is actually a sophisticated hack, the above is both bash _and_ and JavaScript at the same time:

``` js
":" //# comment; exec /usr/bin/env node --harmony --harmony_arrow_functions "$0" "$@"
```

shebanghack

**Comment by MadcapJake** [comment]
So that's staying in, right?  Why is this an issue?

**Comment by ghost** [comment]
@iamstarkov @MadcapJake  [Please see this](http://stackoverflow.com/questions/4303128/how-to-use-multiple-arguments-with-a-shebang-i-e).

> The shebang line has never been specified as part of POSIX, SUS, LSB or any other specification. AFAIK, it hasn't even been properly documented.
> 
> There is a rough consensus about what it does: take everything between the ! and the \n and exec it. The assumption is that everything between the ! and the \n is a full absolute path to the interpreter. There is no consensus about what happens if it contains whitespace.
> - 1　Some operating systems simply treat the entire thing as the path. After all, in most operating systems, whitespace or dashes are legal in a path.
> - 2　Some operating systems split at whitespace and treat the first part as the path to the interpreter and the rest as individual arguments.
> - 3　Some operating systems split at the first whitespace and treat the front part as the path to the interpeter and the rest as a single argument (which is what you are seeing).
> - 4　Some even don't support shebang lines at all.
> 
> Thankfully, 1. and 4. seem to have died out, but 3. is pretty widespread, so you simply cannot rely on being able to pass more than one argument.

**Comment by iamstarkov** [comment]
> Is actually a sophisticated hack, the above is both bash and and JavaScript at the same time:

I’m pretty sure that this hack is not supported on windows platform

**Comment by ghost** [comment]
@iamstarkov We need a new bootstrapping strategy that is also compatible with Windows. User's node version must support generators and ideally arrow functions to play nice with Fly. 

Another thing we could do, is do as you suggested in another issue and everything ES5 files as well. That would make all problems go away, except the starting time will have a ~1 sec delay due to Flyfile compilation.

**Comment by iamstarkov** [comment]
> That would make all problems go away, except the starting time will have a ~1 sec delay due to Flyfile compilation. 

compilation usually happen at prepublish step, not in runtime

**Comment by ghost** [comment]
@iamstarkov This is not accurate.

In Fly, transpilation can happen:
1. At prepublish step, thus Fly shipping precompiled.
2. When loading Flyfiles written in languages other than ES5, for example, Flyfile.babel.js, Flyfile.coffee, Flyfile.eg.

(1) occurs during development using babel and (2) occurs using the require hook supplied by jsVariant if the Flyfile extension is any other than `Flyfile.js`.

**Comment by ghost** [comment]
@iamstarkov 

``` sh
#!/usr/bin/env node --harmony
```

See [this](http://stackoverflow.com/questions/20980222/run-a-node-shell-script-in-harmony-mode).

**Comment by iamstarkov** [comment]
@bucaran fresh 0.4 still not working on windows

**Comment by ghost** [comment]
@iamstarkov yes, there is a bug in `bin/index`. I carelessly used `dist` but this was relative to my project's root, so only noticed now it does not work for a clean install because it can't find the `dist` directory. Will fix this ASAP.

**Comment by ghost** [comment]
@iamstarkov Want to try again? :smile:

**Comment by iamstarkov** [comment]
@bucaran has no time right now, but you can user [modern.ie](http://dev.modern.ie/) image to test work in windows env

**Comment by ghost** [comment]
:+1: Thanks!

**Comment by iamstarkov** [comment]
@bucaran have you tried new fly in windows?

**Comment by ghost** [comment]
@iamstarkov Not really. I was adding tests to all the plugins. Done with mine, and will start PRing others from now.

**Comment by iamstarkov** [comment]
fixed in 0.4

**Comment by ghost** [comment]
Great！(￣-￣)ゞ

**Pull request closed without merging**